src/plugins/data/common/autocomplete_provider/index.ts

```diff
@@ -16,13 +16,22 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+import { AutocompleteProvider } from './types';
 
-const autocompleteProviders = new Map();
+export class AutocompleteProviderRegister {
+  private readonly registeredProviders: Map<string, AutocompleteProvider> = new Map();
 
-export function addAutocompleteProvider(language, provider) {
-  autocompleteProviders.set(language, provider);
-}
+  public addProvider(language: string, provider: AutocompleteProvider): void {
```

Comment:
language should default to `kuery` (this will remove this hardcoded string from ~10 places).

Reply:
I cannot do it =(, We define `kuery` from  the x-pack folder with different licence